### DISCUSSION TITLE
tutorials/lift-off-part2/03-apollo-restdatasource

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#43] [course feedback]
# tutorials/lift-off-part2/03-apollo-restdatasource

https://www.apollographql.com/tutorials/lift-off-part2/03-apollo-restdatasource

**Comment by MauLight** [comment]
all good so far, thanks for this comprehensive course!

**Comment by mc-apolicastro** [comment]
Really enjoying this course so far everything is going great!!

**Comment by belloshehu** [comment]
I have to understand the need for Dataloader in GraphQL backend. Does Apollo server implement it out of the box so we don't have to do it ourselves?

**Reply by lizhennessy** [reply]
Hi there @belloshehu ! Very often, the way we load data is dependent upon the specific data source. Here's an article from the [official Apollo documentation on data loaders with Apollo Server](https://www.apollographql.com/docs/apollo-server/data/fetching-data/#batching-and-caching) that might help you answer your question!

**Reply by belloshehu** [reply]
Thanks a lot @lizhennessy for the response and the link for additional resource.

**Comment by repiemm** [comment]
Neat!, Even when working with multiple data sources, if Apollo can do the cache management and deduplication proccess, it's already worthy even as a middleware.

**Comment by nas03** [comment]
I learned that GraphQL is a abstraction class between Client and Server right? ( Client -> GraphQL -> Server -> Database). However, i don't understand how can i fetch the data  i need with "fetch" without using GraphQL to interact with the database. For example:
I make a HTTP GET request with: 
     const response  = await  fetch(http://localhost:4000/graphQL/users)
then in the api/controllers
      const getUser = (req, res) => {
      //get users from database
}. 
I dont know if i should use graphQL resolvers here to get users from database. Can you explain for me?

**Comment by NishantCoder108** [comment]
## Apollo RestDataSource Lesson

### Key Takeaways🚀
- When working with a simple RESTful API , it's important to avoid making unnecessary calls. For example, if we make one call and receive a list of tracks, we shouldn't make additional calls for the author's track list,  especially if the author is the same.

- Resolvers can assist in efficiently populating only the necessary data  on the client side , reducing the need for multiple calls.

- GraphQL can manage various data sources , including third-party APIs or GraphQL data sources , by providing different endpoints.

**Comment by gustvoassuncao** [comment]
Really enjoying this course!

**Comment by sphaxn** [comment]
The diagrams showing the communication between the GraphQL server and the REST API are really hard to read.  The lettering is crude and almost illegible.  Using a clean, printed typeface instead would make the diagrams much easier to read and understand.
